src/scalib/preprocessing/quantization.py

```diff
@@ -0,0 +1,85 @@
+import numpy as np
+
+
+class QUANTIZER:
+    r"""Quantize a side channel traces given as an array of float into an array of int16.
+        The quantizer estimates a shift and scale that minimize the loss due to the rounding operation.
+
+    .. math::
+        \mathrm{Quantize}( x) = (x - \mathrm{Shift}) \cdot \mathrm{Scale}
+
+    The shift and scale are computed using `n` samples as
+
+    .. math::
+        \mathrm{Shift} = \frac{1}{2} (\max_{i=1}^n x_i + \min_{i=1}^n x_i) \qquad and \qquad  \mathrm{Scale} = \frac{2^{14}}{\max_{i=1}^n x_i - \min_{i=1}^n x_i}.
+
+
+    Parameters
+    ----------
+    ns : int
+        Number of samples in a single trace.
+    shift : np.ndarray[np.float64]
+        The value to shift every traces.
+    scale : np.ndarray[np.float64]
+        The value to scale every traces.
+
+    Examples
+    --------
+    >>> from scalib.preprocessing import QUANTIZER
+    >>> import numpy as np
+    >>> # 500 traces of 200 points
+    >>> noisy_traces : np.ndarray[np.float64] = np.random.randn(500,200)
+    >>> quantizer = QUANTIZER(200, np.zeros(200),np.ones(200))
+    >>> quantizer.fit_shift_scale(noisy_traces)
+    >>> quantized_traces : np.ndarray[np.int16] = quantizer.quantize(noisy_traces)
+    >>> # Can be reused directly on 5000 new traces for instance
+    >>> noisy_traces : np.ndarray[np.float64] = np.random.randn(5000,200)
+    >>> quantized_traces : np.ndarray[np.int16] = quantizer.quantize(noisy_traces)
+    """
+
+    def __init__(
+        self, ns: int, shift: np.ndarray[np.float64], scale: np.ndarray[np.float64]
+    ):
+        self._ns = ns
+        self._shift = shift
+        self._scale = scale
+
+    def fit_shift_scale(self, noisy_traces: np.ndarray[np.float64]) -> None:
```

Comment:
This can be a `@classmethod` that directly creates a new object.

Reply:
Ok done.